exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -182,10 +183,24 @@ public ParquetGroupScan( //
     }
 
     this.selectionRoot = selectionRoot;
-    if (selection instanceof ParquetFileSelection) {
-      final ParquetFileSelection pfs = ParquetFileSelection.class.cast(selection);
-      this.parquetTableMetadata = pfs.getParquetMetadata();
+
+    FileSelection newSelection = null;
+    if (!selection.isExpanded()) {
+      FileStatus firstPath = selection.getFirstPath(fs);
+      Path p = new Path(firstPath.getPath(), Metadata.METADATA_FILENAME);
+      if (!fs.exists(p)) { // no metadata cache
+        if (selection.checkedForDirectories() && selection.hasDirectories()) {
```

Comment:
This won't work if `checkedForDirectories==false`, right ?
`hasDirectories()` already uses `checkedForDirectories` internally, the following should work:
 
    if (selection.hasDirectories()) {